modules/p2/ding_list/theme/theme.inc

```diff
@@ -164,24 +165,33 @@ function theme_ding_list_lists($variables) {
       }
     }
 
-    $out .= theme('item_list', array(
-      'items' => $rows,
-      'title' => $group['title'],
-      'type' => 'ul',
-      'attributes' => array(),
-    ));
+    if (!empty($rows)) {
+      $render = array(
+        '#theme' => 'item_list',
+        '#title' => $group['title'],
+        '#items' => $rows,
+        '#title' => $group['title'],
+        '#type' => 'ul',
+        '#attributes' => array(),
+      );
+      $out .= drupal_render($render);
+    }
   }
 
   // Footer links.
-  $out .= theme('item_list', array(
-    'items' => array(
-      l(t("Mark all notifications as read"), "ding_message/purge_user_messages", array('attributes' => array('class' => array('button')))),
-      l(t("Create new list"), "list/create", array('attributes' => array('class' => array('button')))),
-    ),
-    'title' => "",
-    'type' => 'ul',
-    'attributes' => array('class' => array('bottom-links')),
-  ));
+  if ($variables['uid'] == $user->uid) {
+    $render = array(
+      '#theme' => 'item_list',
+      '#items' => array(
+        l(t("Mark all notifications as read"), "ding_message/purge_user_messages", array('attributes' => array('class' => array('button')))),
+        l(t("Create new list"), "list/create", array('attributes' => array('class' => array('button')))),
+      ),
+      '#title' => "",
+      '#type' => 'ul',
+      '#attributes' => array('class' => array('bottom-links')),
+    );
+    $out .= drupal_render($render);
```

Comment:
It's called from inside a theme_HOOK() function, we could move the creation of the item_list to a preprocessor, but we'd still need to call drupal_render inside the theme_HOOK().